scale-tests/streaming_test.py

```diff
@@ -0,0 +1,115 @@
+    --kerberos-flag <bool>       is Kerberos enabled? [default: false]
+    --num-consumers <n>          number of consumers to create for each producer [default: 1]
+"""
+
+
+import logging
+import json
+
+import sdk_cmd
+import spark_utils
+
+from docopt import docopt
+
+log = logging.getLogger(__name__)
+PRODUCER_NUM_WORDS_PER_MIN = 480
+
+
+def main(dispatchers, jar_url, kafka_pkg_name, kafka_svc_name, num_consumers, desired_runtime, kerberos_flag):
+
+    def _kafka_broker_dns():
+        cmd = "{package_name} --name={service_name} endpoints broker".format(
+            package_name=kafka_pkg_name,
+            service_name=kafka_svc_name)
+        rt, stdout, _ = sdk_cmd.run_raw_cli(cmd)
+        assert rt == 0, "Failed to get broker endpoints"
+        return json.loads(stdout)["dns"][0]
+
+    def _submit_producer(broker_dns, common_conf, topic, spark_app_name, driver_role):
+        big_file = "file:///mnt/mesos/sandbox/big.txt"
+
+        producer_args = " ".join([broker_dns, big_file, topic, kerberos_flag])
+
+        producer_config = ["--conf", "spark.cores.max=2",
+                           "--conf", "spark.executor.cores=2",
+                           "--class", "KafkaFeeder"] + common_conf
+
+        spark_utils.submit_job(app_url=jar_url,
+                               app_args=producer_args,
+                               app_name=spark_app_name,
+                               args=producer_config,
+                               driver_role=driver_role,
+                               verbose=False)
+
+    def _submit_consumer(broker_dns, common_conf, topic, spark_app_name, driver_role, num_words):
+        consumer_args = " ".join([broker_dns, topic, num_words, kerberos_flag])
+
+        consumer_config = ["--conf", "spark.cores.max=4",
+                           "--class", "KafkaConsumer"] + common_conf
+
+        spark_utils.submit_job(app_url=jar_url,
+                               app_args=consumer_args,
+                               app_name=spark_app_name,
+                               args=consumer_config,
+                               driver_role=driver_role,
+                               verbose=False)
+
+    common_conf = [
+        "--supervise",
+        "--conf", "spark.mesos.containerizer=mesos",
+        "--conf", "spark.mesos.driver.failoverTimeout=30",
+        "--conf", "spark.mesos.executor.docker.image=mesosphere/spark-dev:f5dd540adffd9ab9e3e826e48d22e39ebc296567-1d7926a8b500d0105b80a6bb808a671b047dc963",
+        "--conf", "spark.mesos.uris=http://norvig.com/big.txt",
+        "--conf", "spark.port.maxRetries=32",
+        "--conf", "spark.scheduler.maxRegisteredResourcesWaitingTime=2400s",
+        "--conf", "spark.scheduler.minRegisteredResourcesRatio=1.0"
+    ]
+
+    broker_dns = _kafka_broker_dns()
+    num_words_to_read = int(desired_runtime * PRODUCER_NUM_WORDS_PER_MIN)
+
+    for dispatcher in dispatchers:
+        spark_app_name, driver_role = dispatcher.split(",")
```

Comment:
Should this be `spark_app_name, dispatcher_role, driver_role`, or should `dispatcher_role` be removed from the output of `deploy-dispatchers.py`? cc @susanxhuynh

Reply:
I actually had it with the `dispatcher role` in the middle but have seen different variations of whether to use it or not. This seemed to be the latest working version since the Marathon batch app points to a `dispatchers.out` file with only `dispatcher-name`, and `driver-role`.

Reply:
@sam It should be okay now to include all three. The "deploy-dispatchers" script now generates all three in its output file.

Reply:
okay i'll add a placeholder for it since the script itself doesn't use the dispatcher-role